### DISCUSSION TITLE
Direct exports from `styled-system` in `@twilio-paste/styling-library`?

### DISCUSSION THREAD

**Discussion by hharnisc** [twilio-labs/paste#601] [General] [answered]
I see `compose`, `layout` and `space` exports from `@twilio-paste/styling-library` consumed here in the Anchor: https://github.com/twilio-labs/paste/blob/main/packages/paste-core/components/anchor/src/index.tsx#L3

However if I try to consume those parameters in TS based library, the TS compiler complains that the parameters are not exported. Looking into where they are exported, theres a `*` export: https://github.com/twilio-labs/paste/blob/main/packages/paste-libraries/styling/src/index.tsx#L12

Would you be open to exporting the `styled-system` exports directly with something like:

```js
export { compose, layout, space /*any other consumed exports */ } from 'styled-system';
```

That way TS based projects can pick up these exports? Added context: https://code.hq.twilio.com/twilio/console/pull/29/files#diff-a6d06ff156faa73cda9da98d7d5b5d84R5-R11

**Comment by SiTaggart** [comment, marked as answer]
Hmmm I think this might not actually be a problem with the exports as much as where the types are coming from. We have a dev dependency on `@types/styled-system`, but we don't include it as a dependency for the package.

On a simple repro of your exact example, just by adding `@types/styled-system` as an installed dependency made the errors of missing exports disappear. https://codesandbox.io/s/styling-library-exports-test-8gtph?file=/src/index.tsx

For now I think you can just add it as a dep to your project and hopefully it'll remove the need for the ts-ignore's. We'll probably need to add the types package as a dep to the styling-library though in the future. I'll create a ticket for it.

Let us know if adding the types package doesn't solve the issue.

**Reply by hharnisc** [reply]
Ah yep! Good catch, that fixes the issue 🙌
